crates/bevy_tasks/src/slice.rs

```diff
@@ -1,6 +1,35 @@
 use super::TaskPool;
 
+/// Provides functions for mapping read-only slices across a provided [`TaskPool`].
 pub trait ParallelSlice<T: Sync>: AsRef<[T]> {
+    /// Splits the slice in chunks of size `chunks_size` or less and maps the chunks
+    /// in parallel across the provided `task_pool`. One task is spawned in the task pool
+    /// for every chunk.
+    ///
+    /// Returns a `Vec` of the mapped results.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # use bevy_tasks::prelude::*;
+    /// # use bevy_tasks::TaskPool;
+    /// let task_pool = TaskPool::new();
+    /// let counts = (0..10000).collect::<Vec<u32>>();
+    /// let incremented = counts.par_chunk_map(&task_pool, 100, |chunk| {
+    ///   let mut results = Vec::new();
+    ///   for count in chunk {
+    ///     results.push(*count + 2);
+    ///   }
+    ///   results
+    /// });
+    /// # let flattened: Vec<_> = incremented.into_iter().flatten().collect();
+    /// # assert_eq!(flattened, (2..10002).collect::<Vec<u32>>());
+    /// ```
+    ///
+    /// # See Also
+    ///
+    /// [`ParallelSliceMut::par_chunk_map_mut`] for mapping mutable slices.
+    /// [`ParallelSlice::par_splat_map`] for mapping when a specific chunk size is unknown.
```

Comment:
```suggestion
    /// - [`ParallelSliceMut::par_chunk_map_mut`] for mapping mutable slices.
    /// - [`ParallelSlice::par_splat_map`] for mapping when a specific chunk size is unknown.
```

Looks better as a list IMHO. same below :)